File: problem12/p12.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int divisor_count(int number){
    int i = 1, count = 0;

    for(i=1; i<=sqrt(number); i++){
        if(number % i == 0)
            count += 2;
        if(number == i * i)
            count--;
    }

    return count;
}

int find_triangle_num(int limit){

    int i, count = 0;

    while(1){
        count += i;
        i++;
        if(divisor_count(count) > limit)
            break;
    }

    return count;

}

int main()
{

    printf("%d\n", find_triangle_num(500));

    return 0;
}

File: problem12/p12.java
public class p12 {

	public static int divisor_count(int number) {
		
		int i, count = 0;
		
		for(i=1; i<=Math.sqrt(number); i++) {
			if(number % i == 0)
				count += 2;
			if(number == i * i)
				count--;	
		}
		
		return count;
		
	}
	
	public static int find_triangle_num(int limit) {
		
		int i = 1, triangle = 0;
		
		while(true) {
			triangle += i;
			if(divisor_count(triangle) > 500)
				break;
			i++;
		}
		
		return triangle;
		
	}
	
	public static void main(String args[]) {
		
		System.out.println(find_triangle_num(500));
		
	}
	
}
